generate_task_xml.py

```diff
@@ -0,0 +1,103 @@
+import getpass
+import os
+import subprocess
+from datetime import datetime
+
+# 取得當前使用者的名稱
+current_user = getpass.getuser()
+
+
+# 取得當前使用者的SID
+def get_user_sid(username):
+    try:
+        output = subprocess.check_output(
+            f'wmic useraccount where name="{username}" get sid', shell=True
+        )
+        sid = output.decode().split("\n")[1].strip()
+        return sid
+    except Exception as e:
+        print(f"Error getting SID for user {username}: {e}")
+        return None
+
+
+current_user_sid = get_user_sid(current_user)
+computer_name = os.environ["COMPUTERNAME"]
+
+# Let the user choose between Miniconda and Anaconda
+print("Please choose your Python distribution:")
+print("1. Miniconda")
+print("2. Anaconda")
+choice = input("Enter your choice (1 or 2): ")
+
+if choice == "1":
+    base_path = f"C:\\Users\\{current_user}\\miniconda3"
+elif choice == "2":
+    base_path = f"C:\\Users\\{current_user}\\anaconda3"
+else:
+    print("Invalid choice, please rerun the script and choose 1 or 2.")
+    exit(1)
+
+# Let the user enter the name of the environment they are using
+env_name = input("Please enter the name of the environment you are using: ")
+
+# Use the current working directory as the output path
+output_path = os.path.join(os.getcwd(), "MS_reward.xml")
```

Comment:
We should[ get the directory of the script being run](https://stackoverflow.com/questions/3430372/how-do-i-get-the-full-path-of-the-current-files-directory#:~:text=Python%203,__file__).parent.resolve()) versus the cwd to be a little more flexible in case someone runs this script from a different directory than project root.